desktop/app/installer.js

```diff
@@ -1,65 +1,32 @@
 import {app} from 'electron'
-import {exec} from 'child_process'
-import path from 'path'
-import fs from 'fs'
-import os from 'os'
+import {appInstallerPath, appBundlePath} from './paths'
+import exec from './exec'
 
 import {runMode} from '../shared/constants/platform.native.desktop'
 
 export default callback => {
-  const appPath = app.getAppPath()
-  // For testing when running manually via npm start
-  // const appPath = '/Applications/Keybase.app/Contents/Resources/app/'
-  // For testing running from DMG
-  // const appPath = '/Volumes/Keybase/Keybase.app/Contents/Resources/app/'
-
-  const resourcesPath = path.resolve(appPath, '..')
-  const appBundlePath = path.resolve(appPath, '..', '..', '..')
-  const installerExec = path.resolve(resourcesPath, 'KeybaseInstaller.app', 'Contents', 'MacOS', 'Keybase')
-
-  const platform = os.platform()
-  if (platform !== 'darwin') {
-    callback(null)
+  const installerPath = appInstallerPath()
+  if (!installerPath) {
+    callback(new Error('No installer path'))
     return
   }
+  const bundlePath = appBundlePath()
+  if (!bundlePath) {
+    callback(new Error('No bundle path for installer'))
+    return
+  }
+  const args = ['--app-path=' + bundlePath, '--run-mode=' + runMode]
```

Comment:
Could shell-escape/quote the variable, or use `spawn` which doesn't use the shell. Again: I don't see how this could be abused, but wanted to raise awareness in case someone with greater background on this spotted one.